src/main/groovy/com/lesfurets/jenkins/unit/BasePipelineTest.groovy

```diff
@@ -189,8 +189,16 @@ abstract class BasePipelineTest {
                 assert item.size() == 2, "withEnv list does not look right: ${list.toString()}"
                 addEnvVar(item[0], item[1])
             }
-            c.delegate = binding
-            c.call()
+            try {
+                c.delegate = binding
+                c.call()
```

Comment:
save/restore env looks to be a natural approach, also it needs additional testcase

Reply:
What would you do in addition? The test case I added already confirms that the environment values are not present after the restore. Please suggest what is missing.

Reply:
I mean to add a testcase to cover nested `withEnv`. Same as I described in #247. Instead of removing a field from `env` map after `c.call()` you just need to save/restore entire `env`.

Reply:
Is there any use case of the environment getting modified by the inner closure and restoring the entire env would be invalid? I am not sure Jenkins supports that but I will test to make sure it works as we think. Easy to add a full save, restore the env instead of the second iteration.

Reply:
Added full save/restore of env value